source/grid/grid_tools.cc

```diff
@@ -1201,14 +1202,8 @@ namespace GridTools
             }
         }
 
-    // create a random number generator for the interval [-1,1]. we use
-    // this to make sure the distribution we get is repeatable, i.e.,
-    // if you call the function twice on the same mesh, then you will
-    // get the same mesh. this would not be the case if you used
-    // the rand() function, which carries around some internal state
-    // we could use std::mt19937 but doing so results in compiler-dependent
-    // output.
```

Comment:
Would you mind re-instating this comment? I think it describes relevant behavior.

Reply:
The reason why I removed the comment is that it says that the result is reproducible when the function is applied over the same mesh since it uses mt19937 instead of rand(). However, that behavior is not given by mt19937 instead of rand(), but because it uses mt19937 with its default initialization, which happens to be deterministic (i.e., has a default seed). 

With the new seed parameter, that piece of comment is outdated since obtaining or not the same result is defined by the value of the seed, over which the user has control in the function call.

If you think this clarification should be added to the code, let me know.

Reply:
Right. I think what I want to say is that "with the same seed, the function produces the same reproducible output", which is not true when using `rand()`. But let's not worry about it. I can always re-introduce the comment later on in slightly modified form.